### PR TITLE
Fading into alarm sound.

### PR DESCRIPTION
**Pull request by mjtice**
I figured I can't be the only one who hates getting scared out of sleep.  Essentially when `fade` is true then volume is set to zero and then it is incremented `fade_step` amount each second until `fade_timer` is reached.  Then volume is set to `volume`.  It was a little tricky figuring out a good `fade_step` until I settled on .5%.

### CODE REVIEW

File: MMM-AlarmClock.js [hunk 1]
```diff
@@ -73,6 +76,28 @@ Module.register('MMM-AlarmClock', {
         }
     },
 
+    fadeAlarm() {
+        var max_volume = this.config.volume;
+        var volume = 0;
+        var that = this;
```
Comment (fewieden) [inline MMM-AlarmClock.js]: no need for that, while using an arrow function

File: MMM-AlarmClock.js [hunk 2]
```diff
@@ -73,6 +76,28 @@ Module.register('MMM-AlarmClock', {
...
+        var volume_step = this.config.fade_step;
+        var counter = 0;
+        var myTimer = setInterval(() => {
+            document.getElementById('alarm_player').volume = volume;
+            volume = volume + volume_step;
+            console.log(volume);
+            counter = counter + 1000;
+            if (volume >= max_volume) {
+                document.getElementById(that.identifier).volume = max_volume;
```
Comment (fewieden) [inline MMM-AlarmClock.js]: the identifier of the module is not the same as the id you used in the other parts

File: MMM-AlarmClock.js [hunk 3]
```diff
@@ -73,6 +76,28 @@ Module.register('MMM-AlarmClock', {
...
     },
 
+    fadeAlarm() {
+        var max_volume = this.config.volume;
+        var volume = 0;
+        var that = this;
+        var volume_step = this.config.fade_step;
+        var counter = 0;
+        var myTimer = setInterval(() => {
+            document.getElementById('alarm_player').volume = volume;
```
Comment (fewieden) [inline MMM-AlarmClock.js]: please use the module namespace when using ids like MMM-AlarmClock-Player

File: MMM-AlarmClock.js [hunk 4]
```diff
@@ -149,6 +175,9 @@ Module.register('MMM-AlarmClock', {
             sound.volume = this.config.volume;
             sound.setAttribute('autoplay', true);
             sound.setAttribute('loop', true);
+            if (this.config.fade == true) {
+                this.fadeAlarm(sound);
```
Comment (fewieden) [inline MMM-AlarmClock.js]: you're not using the parameter in the function itself

File: MMM-AlarmClock.js [hunk 5]
```diff
@@ -141,6 +166,7 @@ Module.register('MMM-AlarmClock', {
             wrapper.appendChild(text);
         } else if (this.alarmFired) {
             const sound = document.createElement('audio');
+            sound.setAttribute('id', 'alarm_player');
```
Comment (fewieden) [inline MMM-AlarmClock.js]: see namespace above

File: MMM-AlarmClock.js [hunk 6]
```diff
@@ -73,6 +76,28 @@ Module.register('MMM-AlarmClock', {
...
+    fadeAlarm() {
+        var max_volume = this.config.volume;
+        var volume = 0;
+        var that = this;
+        var volume_step = this.config.fade_step;
+        var counter = 0;
+        var myTimer = setInterval(() => {
+            document.getElementById('alarm_player').volume = volume;
+            volume = volume + volume_step;
+            console.log(volume);
```
Comment (fewieden) [inline MMM-AlarmClock.js]: this debug log will spam the logs

**Comment by mjtice** [comment]
I finally made some time to get these changes made.  I've updated my fork (I think that's all you need?).  I'm not sure about the travis stuff however.

**Comment by fewieden** [comment]
Thanks for updating. Travis has issues with the grey highlited parts https://travis-ci.org/fewieden/MMM-AlarmClock/jobs/221531319#L489-L506

```
   23:20  error  A leading decimal point can be confused with a dot   no-floating-decimal

   80:9   error  Unexpected var, use let or const instead             no-var

   80:13  error  Identifier 'max_volume' is not in camel case         camelcase

   81:9   error  Unexpected var, use let or const instead             no-var

   82:9   error  Unexpected var, use let or const instead             no-var

   82:13  error  Identifier 'volume_step' is not in camel case        camelcase

   83:9   error  Unexpected var, use let or const instead             no-var

   84:9   error  Unexpected var, use let or const instead             no-var

   86:13  error  Assignment can be replaced with operator assignment  operator-assignment

   86:31  error  Identifier 'volume_step' is not in camel case        camelcase

   87:13  error  Assignment can be replaced with operator assignment  operator-assignment

   88:27  error  Identifier 'max_volume' is not in camel case         camelcase

   89:75  error  Identifier 'max_volume' is not in camel case         camelcase

   93:75  error  Identifier 'max_volume' is not in camel case         camelcase

  176:34  error  Expected '===' and instead saw '=='                  eqeqeq
```

I will take care about it this weekend

**Pull request merged**